CefSharp.Example/RequestEventHandler/RequestEventHandler.cs

```diff
@@ -0,0 +1,155 @@
+    ///     <see cref="IRequestHandler.OnCertificateError" />)
+    ///     inspired by:
+    ///     https://github.com/cefsharp/CefSharp/blob/fa41529853b2527eb0468a507ab6c5bd0768eb59/CefSharp.Example/RequestHandler.cs
+    /// </summary>
+    public class RequestEventHandler : IRequestHandler
+    {
+        public event EventHandler<OnBeforeBrowseEventArgs> OnBeforeBrowseEvent;
+        public event EventHandler<OnOpenUrlFromTabEventArgs> OnOpenUrlFromTabEvent;
+        public event EventHandler<OnCertificateErrorEventArgs> OnCertificateErrorEvent;
+        public event EventHandler<OnPluginCrashedEventArgs> OnPluginCrashedEvent;
+        public event EventHandler<OnBeforeResourceLoadEventArgs> OnBeforeResourceLoadEvent;
+        public event EventHandler<GetAuthCredentialsEventArgs> GetAuthCredentialsEvent;
+        public event EventHandler<OnRenderProcessTerminatedEventArgs> OnRenderProcessTerminatedEvent;
+        public event EventHandler<OnQuotaRequestEventArgs> OnQuotaRequestEvent;
+        public event EventHandler<OnResourceRedirectEventArgs> OnResourceRedirectEvent;
+
+        /// <summary>
+        ///     SECURITY WARNING: YOU SHOULD USE THIS EVENT TO ENFORCE RESTRICTIONS BASED ON SCHEME, HOST OR OTHER URL ANALYSIS
+        ///     BEFORE ALLOWING OS EXECUTION.
+        /// </summary>
+        public event EventHandler<OnProtocolExecutionEventArgs> OnProtocolExecutionEvent;
+        public event EventHandler<OnRenderViewReadyEventArgs> OnRenderViewReadyEvent;
+        public event EventHandler<OnResourceResponseEventArgs> OnResourceResponseEvent;
+        public event EventHandler<GetResourceResponseFilterEventArgs> GetResourceResponseFilterEvent;
+        public event EventHandler<OnResourceLoadCompleteEventArgs> OnResourceLoadCompleteEvent;
+
+        public bool OnBeforeBrowse(IWebBrowser browserControl, IBrowser browser, IFrame frame, IRequest request, bool isRedirect)
+        {
+            var args = new OnBeforeBrowseEventArgs(browserControl, browser, frame, request, isRedirect);
+            OnBeforeBrowseEvent?.Invoke(this, args);
+            return args.CancelNavigation;
+        }
+
+        public bool OnOpenUrlFromTab(IWebBrowser browserControl, IBrowser browser, IFrame frame, string targetUrl, WindowOpenDisposition targetDisposition, bool userGesture)
+        {
+            var args = new OnOpenUrlFromTabEventArgs(browserControl, browser, frame, targetUrl, targetDisposition, userGesture);
+            OnOpenUrlFromTabEvent?.Invoke(this, args);
+            return args.CancelNavigation;
+        }
+
+        public bool OnCertificateError(IWebBrowser browserControl, IBrowser browser, CefErrorCode errorCode, string requestUrl, ISslInfo sslInfo, IRequestCallback callback)
+        {
+            var args = new OnCertificateErrorEventArgs(browserControl, browser, errorCode, requestUrl, sslInfo, callback);
+            OnCertificateErrorEvent?.Invoke(this, args);
+
+            if (!args.Callback.IsDisposed)
```

Comment:
I just added 2 private methods `EnsureCallbackDisposal(...)` . I'd have preferred to use a kind of `IDisposalAware` interface as parameter instead of overloading it, but that requires core changes which is just too much here.